lib/paperclip/interpolations.rb

```diff
@@ -29,11 +30,19 @@ def self.all
     # an interpolation pattern for Paperclip to use.
     def self.interpolate pattern, *args
       pattern = args.first.instance.send(pattern) if pattern.kind_of? Symbol
-      all.reverse.inject(pattern) do |result, tag|
-        result.gsub(/:#{tag}/) do |match|
-          send( tag, *args )
-        end
+      result = pattern.dup
+      interpolators_cache.each do |tag|
+        result.gsub!(tag_pattern_cache[tag]) {send( tag, *args )}
       end
+      result
+    end
+
+    def self.tag_pattern_cache
+      @tag_pattern_cache ||= Hash.new {|hash, key| hash[key]= ":#{key}"}
```

Comment:
Surrounding space missing for operator '='.<br>Space between { and | missing.<br>Space missing inside }.